src/main/java/fi/aalto/cs/apluscourses/ui/base/BaseListView.java

```diff
@@ -90,10 +92,16 @@ public void setModel(@Nullable ListModel<E> model) {
     if (model == null) {
       model = new DefaultListModel<>();
     }
-    super.setModel(model);
-    if (model instanceof SelectableListModel) {
-      setSelectionModel(((SelectableListModel<E>) model).getSelectionModel());
-    }
+    ListModel<E> finalModel = model;
+    ApplicationManager.getApplication().invokeLater(
```

Comment:
It's a bit odd that superclass's method is called in another thread. Could `invokeLater` be a responsibility of a caller of this method, instead.

Reply:
Good point! will change